core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3313,15 +3318,22 @@ object Stream extends StreamLowPriority {
     }
   }
 
+  private[fs2] final class PartiallyAppliedFromBlockingIterator[F[_]](
+      private val blocking: Boolean
+  ) extends AnyVal {
+    def apply[A](iterator: Iterator[A], chunkSize: Int)(implicit F: Sync[F]): Stream[F, A] =
+      new PartiallyAppliedFromIterator(blocking).apply(iterator, chunkSize)
+  }
+
   /** Lifts an iterator into a Stream.
     */
   def fromIterator[F[_]]: PartiallyAppliedFromIterator[F] =
     new PartiallyAppliedFromIterator(blocking = false)
 
   /** Lifts an iterator into a Stream, shifting any interaction with the iterator to the blocking pool.
     */
-  def fromBlockingIterator[F[_]]: PartiallyAppliedFromIterator[F] =
-    new PartiallyAppliedFromIterator(blocking = true)
+  def fromBlockingIterator[F[_]]: PartiallyAppliedFromBlockingIterator[F] =
+    new PartiallyAppliedFromBlockingIterator(blocking = true)
```

Comment:
This signature change is binary-compatible because `PartiallyAppliedFromBlockingIterator` is an `AnyVal` for `Boolean`, just like `PartiallyAppliedFromIterator`.